vdev/src/testing/integration.rs

```diff
@@ -135,30 +141,64 @@ impl IntegrationTest {
 }
 
 struct Compose {
-    path: PathBuf,
+    original_path: PathBuf,
     test_dir: PathBuf,
     env: Environment,
     #[cfg(unix)]
     config: ComposeConfig,
     network: String,
+    #[cfg(unix)]
+    temp_file: NamedTempFile,
 }
 
 impl Compose {
     fn new(test_dir: PathBuf, env: Environment, network: String) -> Result<Option<Self>> {
-        let path: PathBuf = [&test_dir, Path::new("compose.yaml")].iter().collect();
-        match path.try_exists() {
-            Err(error) => Err(error).with_context(|| format!("Could not lookup {path:?}")),
+        let original_path: PathBuf = [&test_dir, Path::new("compose.yaml")].iter().collect();
+
+        match original_path.try_exists() {
+            Err(error) => Err(error).with_context(|| format!("Could not lookup {original_path:?}")),
             Ok(false) => Ok(None),
             Ok(true) => {
                 #[cfg(unix)]
-                let config = ComposeConfig::parse(&path)?;
+                let mut config = ComposeConfig::parse(&original_path)?;
+
+                #[cfg(unix)]
+                let temp_file: NamedTempFile;
+
+                #[cfg(unix)]
+                {
+                    // Inject the networks block
+                    config.networks.insert(
+                        "default".to_string(),
+                        BTreeMap::from_iter([("name".to_string(), network.clone())]),
+                    );
+
+                    // Create a named tempfile, there may be resource leakage here in case of SIGINT
+                    // Tried tempfile::tempfile() but this returns a File object without a usable path
+                    // https://docs.rs/tempfile/latest/tempfile/#resource-leaking
+                    temp_file = Builder::new()
+                        .prefix("compose-temp-")
+                        .suffix(".yaml")
+                        .tempfile_in(&test_dir)
+                        .with_context(|| "Failed to create temporary compose file")?;
+
+                    fs::write(
+                        temp_file.path(),
+                        serde_yaml::to_string(&config).with_context(|| {
+                            "Failed to serialize modified compose.yml".to_string()
+                        })?,
+                    )?;
+                }
```

Comment:
Could this all be moved into a function to avoid the awkward `cfg` bits?

Reply:
Marking this as resolved as there is no longer `cfg` bits in this code block